Improve contributing docs (#387)
oxidecomputer/terraform-provider-oxide: CONTRIBUTING.md

`make build` will only build the binary in the `./bin` directory. Terraform will not know to look for the provider there, and will not work with Terraform configuration files.

### Building with local SDK changes or other versions

> If you wish to try out the provider with changes in your local SDK, all you need to do is run `make local-api`. The make target assumes you have  the `oxide.go`
> inside the same parent directory as this one (e.g. your-dir/oxide.go  and your-dir/terraform-provider-oxide).

Reviewer: ````suggestion
To use the Terraform provider with a local `oxide.go` Go SDK run `make local-api`.
This target assumes both the `oxide.go` and `terraform-provider-oxide` repositories
are checked out to adjacent directories (e.g., share the same parent directory).

```
.
├── oxide.go
└── terraform-provider-oxide
```
````

To undo those changes run `make unset-local-api`.

> If the SDK you wish to try out with another version or a Go pseudo-version of the SDK run `SDK_V={GIT_HASH|VERSION} make sdk-version`.

Reviewer: ```suggestion
To use a specific version of the Go SDK run `SDK_V={GIT_HASH|VERSION} make sdk-version`.
```
